src/configurationGenerator.ts

```diff
@@ -0,0 +1,99 @@
+import * as Globals from './constants'
+import {Shot} from './shot'
+import {BoardState} from './boardState'
+import {Position} from './position'
+import {ShipPlacement} from './shipPlacement'
+
+export class ConfigurationGenerator{
```

Comment:
I'm much happier with this class, in general. It's carrying a sensible amount of logic, all of the methods are cut into easily parsed pieces, everything is well named, etc.